src/views/Drawer/DrawerView.js

```diff
@@ -49,111 +47,127 @@ export default class DrawerView extends React.PureComponent {
     if (routes[index].routeName !== drawerCloseRoute) {
       this.props.navigation.navigate(drawerCloseRoute);
     }
   };
 
-  _updateScreenNavigation = navigation => {
-    const { drawerCloseRoute } = this.props;
-    const navigationState = navigation.state.routes.find(
-      route => route.routeName === drawerCloseRoute
-    );
-    if (
-      this._screenNavigationProp &&
-      this._screenNavigationProp.state === navigationState
-    ) {
-      return;
-    }
-    this._screenNavigationProp = addNavigationHelpers({
-      dispatch: navigation.dispatch,
-      state: navigationState,
-      addListener: getChildEventSubscriber(
-        navigation.addListener,
-        navigationState.key
-      ),
-    });
-  };
-
   _updateWidth = () => {
     const drawerWidth =
-      typeof this.props.drawerWidth === 'function'
-        ? this.props.drawerWidth()
-        : this.props.drawerWidth;
+      typeof this.props.navigationConfig.drawerWidth === 'function'
+        ? this.props.navigationConfig.drawerWidth()
+        : this.props.navigationConfig.drawerWidth;
 
     if (this.state.drawerWidth !== drawerWidth) {
       this.setState({ drawerWidth });
     }
   };
 
   _getNavigationState = navigation => {
-    const { drawerCloseRoute } = this.props;
+    const { drawerCloseRoute } = this.props.navigationConfig;
     const navigationState = navigation.state.routes.find(
       route => route.routeName === drawerCloseRoute
     );
+
     return navigationState;
   };
 
-  _renderNavigationView = () => (
-    <DrawerSidebar
-      screenProps={this.props.screenProps}
-      navigation={this._screenNavigationProp}
-      router={this.props.router}
-      contentComponent={this.props.contentComponent}
-      contentOptions={this.props.contentOptions}
-      drawerPosition={this.props.drawerPosition}
-      style={this.props.style}
-    />
-  );
+  _renderNavigationView = () => {
+    const details = Object.values(this.props.descriptors).find(
+      d => d.state.routeName === this.props.navigationConfig.drawerCloseRoute
+    );
 
-  render() {
-    const DrawerScreen = this.props.router.getComponentForRouteName(
-      this.props.drawerCloseRoute
+    const router = details.getComponent().router;
+    const { state, addListener, dispatch } = this.props.navigation;
+    const { routes } = details.state;
+
+    const tabDescriptors = {};
+    routes.forEach(route => {
+      const getComponent = () =>
+        router.getComponentForRouteName(route.routeName);
+
+      const childNavigation = addNavigationHelpers({
+        dispatch,
+        state: route,
+        addListener: getChildEventSubscriber(addListener, route.key),
+      });
+      const options = router.getScreenOptions(
+        childNavigation,
+        this.props.screenProps
+      );
+      tabDescriptors[route.key] = {
+        key: route.key,
+        getComponent,
+        options,
+        state: route,
+        navigation: childNavigation,
+      };
+    });
+
```

Comment:
somewhere here the `navigationConfig` for drawer icons isn't being threaded through correctly, I believe. you can repro by going to the drawer example in playground and seeing that there are no icons, but in the one published from master there are